packaging/packager

```diff
@@ -56,41 +56,35 @@ if ! type zip > /dev/null 2>&1; then
     exit 1
 fi
 
-function pluck_so_files() {
+function find_so_files() {
     sed -E '/\.so$|\.so\.[0-9]+$/!d'
 }
 
 function package_libc_alpine() {
-    if grep --fixed-strings "Alpine Linux" < /etc/os-release > /dev/null; then
+    if grep -F "Alpine Linux" < /etc/os-release > /dev/null; then
         if type apk > /dev/null 2>&1; then
-            apk info --contents musl 2>/dev/null | pluck_so_files | sed 's/^/\//'
+            apk info --contents musl 2>/dev/null | find_so_files | sed 's/^/\//'
         fi
     fi
 }
 
 function package_libc_pacman() {
     if grep --extended-regexp "Arch Linux|Manjaro Linux" < /etc/os-release > /dev/null 2>&1; then
         if type pacman > /dev/null 2>&1; then
-            pacman --query --list --quiet glibc | pluck_so_files
+            pacman --query --list --quiet glibc | find_so_files
         fi
     fi
 }
 
 function package_libc_dpkg() {
     if type dpkg-query > /dev/null 2>&1; then
-        if [[ $(dpkg-query --listfiles libc6 | wc -l) -gt 0 ]]; then
-            dpkg-query --listfiles libc6 | pluck_so_files
-        fi
+        dpkg-query --listfiles libc6:$(dpkg --print-architecture) | find_so_files
     fi
 }
 
 function package_libc_rpm() {
-    arch=$(uname -m)
-
     if type rpm > /dev/null 2>&1; then
-       if [[ $(rpm --query --list glibc.$arch | wc -l) -gt 1 ]]; then
-           rpm --query --list glibc.$arch | pluck_so_files
-       fi
+        rpm --query --list glibc.$(uname -m) | find_so_files
```

Comment:
Same as above. Don't change what is already working.

Reply:
I'd still like to propose that the simpler version is more robust. Maybe we can test it to make sure it works.

Reply:
Have you tested Amazon Linux, Arch Linux, Manjaro, etc.?

I don't see the benefits worth the risk here. If you're right, it still works the same, and if you're wrong we end up with a lot of PR churn and less confidence in the project overall. I'm sorry.

Please just fix the dpkg arch part and revert the rest.

Reply:
Personally I would merge this after a careful review.

Reply:
Perhaps this PR can stay open so people can test it out.

Reply:
> Have you tested Amazon Linux, Arch Linux, Manjaro, etc.?
> 
> I don't see the benefits worth the risk here. If you're right, it still works the same, and if you're wrong we end up with a lot of PR churn and less confidence in the project overall. I'm sorry.
> 
> Please just fix the dpkg arch part and revert the rest.

With the Alpine changes having been cherry-picked into #137, this PR is now well-scoped into only removing the `if [[ ` and `wc -l` guards on "RHEL-like" and "Debian-like" distros, making the logic consistent with "Arch-like" and Alpine. In my understanding, these guards were redundant with the pipelining to `find_so_files`.

I think this is merge-able now with the CI added since this comment @marcomagdy - LMK if you see some other risky aspect.